### PR TITLE
 test/rgw/logging: add bucket logging test suite to teuthology

### PR DESCRIPTION
**Pull request by yuvalif**
this is depended with this s3test PR: https://github.com/ceph/s3-tests/pull/576
    
and the bucket logging PR: https://github.com/ceph/ceph/pull/59808

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [x] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [x] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
- `jenkins test rook e2e`
</details>

### CODE REVIEW

**Comment by yuvalif** [comment]
the bucket-logging test always fail with:
```
yes | sudo mkfs.xfs -f -i size=2048 /dev/vg_nvme/lv_1
mkfs.xfs: /dev/vg_nvme/lv_1 contains a mounted filesystem
```
e.g. https://pulpito.ceph.com/yuvalif-2024-11-14_16:50:25-rgw:bucket-logging-wip-yuval-bucket-logging3-distro-default-smithi/

any idea if this is realted to anything in this PR?

**Comment by yuvalif** [comment]
> the bucket-logging test always fail with:
> 
> ```
> yes | sudo mkfs.xfs -f -i size=2048 /dev/vg_nvme/lv_1
> mkfs.xfs: /dev/vg_nvme/lv_1 contains a mounted filesystem
> ```
> 
> e.g. https://pulpito.ceph.com/yuvalif-2024-11-14_16:50:25-rgw:bucket-logging-wip-yuval-bucket-logging3-distro-default-smithi/
> 
> any idea if this is realted to anything in this PR?

@cbodley i'm getting this error consistently in the test. can this be related to the change here?

**Comment by yuvalif** [comment]
jenkins test windows

**Comment by yuvalif** [comment]
jenkins test api

**Comment by yuvalif** [comment]
jenkins test docs

File: qa/suites/rgw/bucket-logging/tasks/s3tests.yaml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+tasks:
+- install:
+- ceph:
+- rgw: [client.0]
+- tox: [client.0]
```
Comment (cbodley) [inline qa/suites/rgw/bucket-logging/tasks/s3tests.yaml:5]: > @cbodley i'm getting this error consistently in the test. can this be related to the change here?

these tasks are already run as part of `0-install.yaml`, which is why you see `install` trying to run these mkfs.xfs commands a second time

Comment (cbodley) [reply qa/suites/rgw/bucket-logging/tasks/s3tests.yaml:5]: http://qa-proxy.ceph.com/teuthology/yuvalif-2024-11-14_16:50:25-rgw:bucket-logging-wip-yuval-bucket-logging3-distro-default-smithi/7994195/teuthology.log shows the expanded list of tasks near the top:
```
tasks:
  - install: null
  - ceph: null
  - openssl_keys: null
  - rgw:
    - client.0
  - tox:
    - client.0
  - install: null
  - ceph: null
  - rgw:
    - client.0
  - tox:
    - client.0
  - s3tests:
      client.0:
        boto3_extensions: true
        extra_attrs:
        - bucket_logging
        rgw_server: client.0
```

**Comment by yuvalif** [comment]
@cbodley now i get this error:
```
ValueError: Could not determine what branch to use for s3-tests. Please add 'force-branch: {s3-tests branch name}' to the .yaml config for this s3tests task.
```
it says it is missing from the client config.
i set the s3test branch and repo here: https://github.com/ceph/ceph/pull/60700/files#diff-895f52dabf9b3c55f2658e7000786d78d7b68e77860bdda4973c8b892df0c65f

is this the wrong place?

**Comment by cbodley** [comment]
> is this the wrong place?

you'll need to add a symlink for s3tests-branch.yaml under rgw/bucket-logging/. that's how other subsuites work:
```
~/ceph/qa/suites/rgw/verify $ ll s3tests-branch.yaml 
lrwxrwxrwx. 1 cbodley cbodley 27 Dec 14  2023 s3tests-branch.yaml -> .qa/rgw/s3tests-branch.yaml
```

**Comment by yuvalif** [comment]
jenkins test make check

**Comment by yuvalif** [comment]
jenkins test windows

**Comment by yuvalif** [comment]
bucket logging teuthology tests are passing: https://pulpito.ceph.com/yuvalif-2024-11-21_18:53:54-rgw:bucket-logging-wip-yuval-bucket-logging3-distro-default-smithi/
rgw regression: https://pulpito.ceph.com/yuvalif-2024-11-21_18:54:38-rgw-wip-yuval-bucket-logging3-distro-default-smithi/
has failures with multiste: test_bucket_sync_disable, test_bucket_reshard_incremental
and java tests (a know issue)

File: qa/rgw/s3tests-branch.yaml [hunk 2]
```diff
@@ -1,4 +1,4 @@
 overrides:
   s3tests:
-    force-branch: ceph-master
-    # git_remote: https://github.com/ceph/
+    force-branch: wip-yuval-bucket-logging 
+    git_remote: https://github.com/yuvalif/
```
Comment (cbodley) [inline qa/rgw/s3tests-branch.yaml:4]: please remove before merging

**Comment by yuvalif** [comment]
make test keep failing on:
```
ceph/deployment/service_spec.py:1642:12: E721 do not compare types, for exact checks use `is` / `is not`, for instance checks use `isinstance()`
ceph/deployment/service_spec.py:1648:12: E721 do not compare types, for exact checks use `is` / `is not`, for instance checks use `isinstance()`
ceph/deployment/service_spec.py:1654:12: E721 do not compare types, for exact checks use `is` / `is not`, for instance checks use `isinstance()`
ceph/deployment/service_spec.py:1660:12: E721 do not compare types, for exact checks use `is` / `is not`, for instance checks use `isinstance()`
lint: exit 1 (2.01 seconds) /home/jenkins-build/build/workspace/ceph-pull-requests/src/python-common> flake8 ceph pid=3431996
lint: FAIL ✖ in 22.45 seconds
```

**Comment by cbodley** [comment]
> make test keep failing on:

tracked in https://tracker.ceph.com/issues/69024

**Comment by yuvalif** [comment]
commit moved to main PR: https://github.com/ceph/ceph/pull/59808

**Pull request closed without merging**